lib/Raven/Client.php

```diff
@@ -127,6 +136,16 @@ class Client
      */
     private $pendingRequests = [];
 
+    /**
+     * @var callable The tip of the middleware call stack
+     */
+    private $middlewareTip;
```

Comment:
Maybe we should call it `firstMiddleware` or `middlewareStackTip` for more clarity?